fuel-vm/src/interpreter/gas.rs

```diff
@@ -21,37 +27,102 @@ impl<S, Tx> Interpreter<S, Tx> {
     }
 
     pub(crate) fn dependent_gas_charge(&mut self, gas_cost: DependentCost, arg: Word) -> Result<(), RuntimeError> {
-        if gas_cost.dep_per_unit == 0 {
-            self.gas_charge(gas_cost.base)
-        } else {
-            self.gas_charge(gas_cost.base.saturating_add(arg.saturating_div(gas_cost.dep_per_unit)))
-        }
+        let current_contract = self.contract_id();
+        let ReadRegisters { pc, ggas, cgas, is, .. } = split_registers(&mut self.registers).0;
+        let profiler = ProfileGas {
+            pc: pc.as_ref(),
+            is: is.as_ref(),
+            current_contract,
+            profiler: &mut self.profiler,
+        };
+        dependent_gas_charge(cgas, ggas, profiler, gas_cost, arg)
     }
 
     pub(crate) fn gas_charge(&mut self, gas: Word) -> Result<(), RuntimeError> {
+        let current_contract = self.contract_id();
+        let ReadRegisters { pc, ggas, cgas, is, .. } = split_registers(&mut self.registers).0;
```

Comment:
Hmm, do we really need to split it each time?

Reply:
Yes because we need multiple borrows to the registers where at least one borrow is mutable. Also I've benched this and it takes under 1ns vs 3ns if you do it with `split_at_mut`